### PR TITLE
Define presentation compositor

### LINKED ISSUE #58: Define 'Presentation Compositor' [TAG review]
Please forgive my ignorance as I get up to speed with this work.

The IMSC HRM spec contains the term 'Presentation Compositor' several times but does not link to or define what this is. Could you provide or point me to an explanation?

(cc https://github.com/w3ctag/design-reviews/issues/788)

### PR DESCRIPTION
**Pull request by palemieux**
Closes #58

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/imsc-hrm/pull/61.html" title="Last updated on Jan 16, 2023, 11:41 AM UTC (b03fcfc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/imsc-hrm/61/3574995...b03fcfc.html" title="Last updated on Jan 16, 2023, 11:41 AM UTC (b03fcfc)">Diff</a>

### CODE REVIEW

**Comment by palemieux** [comment]
@rhiaro See proposed clarification, which should have made finding the meaning of presentation compositor easier.

File: spec/imsc-hrm.html [hunk 1]
```diff
@@ -425,8 +425,8 @@ <h2>Architecture</h2>
       <a>Intermediate Synchronic Documents</a> E<sub>i</sub> obtained from an input <a>IMSC Document Instance</a>:</p>
       <ul>
         <li><a>non-empty ISDs</a> are processed using a simple double buffering model: while a <a>non-empty ISD</a>
-        E<sub>n</sub> is being painted into a Back Buffer, the previous <a>non-empty ISD</a> E<sub>m</sub> is available
-        for display in a Front Buffer; and</li>
+        E<sub>n</sub> is being painted into a Back Buffer by the Presentation Compositor, the previous <a>non-empty
```
Comment (nigelmegitt) [inline spec/imsc-hrm.html:428]: Seems like we should normatively define "Presentation Compositor"?

Comment (palemieux) [reply spec/imsc-hrm.html:428]: Do you mean with a `dfn`?

Comment (nigelmegitt) [reply spec/imsc-hrm.html:428]: Yes

Comment (palemieux) [reply spec/imsc-hrm.html:428]: Definitions are pretty inconsistent in the document and we already a [separate issue](https://github.com/w3c/imsc-hrm/issues/47) to address that.

File: spec/imsc-hrm.html [hunk 2]
```diff
@@ -433,8 +433,8 @@ <h2>Architecture</h2>
       <a>Intermediate Synchronic Documents</a> E<sub>i</sub>:</p>
       <ul>
         <li><a>non-empty ISDs</a> are processed using a simple double buffering model: while a <a>non-empty ISD</a>
-        E<sub>n</sub> is being painted into a Back Buffer, the previous <a>non-empty ISD</a> E<sub>m</sub> is available
-        for display in a Front Buffer; and</li>
+        E<sub>n</sub> is being painted into a Back Buffer by the <dfn>Presentation Compositor</dfn>, the previous <a>non-empty
+        ISD</a> E<sub>m</sub> is available for display in a Front Buffer; and</li>
```
Comment (nigelmegitt) [inline spec/imsc-hrm.html:437]: Although the term Presentation Compositor is now wrapped in `<dfn>`, it doesn't really help the reader understand what the Presentation Compositor _is_. I was hoping for a definition that would define it at least a little, for example by saying it is a conceptual or hypothetical Presentation Processor assumed to be capable of rendering ISDs from conformant IMSC Document Instances and rasterising the output into an array of pixels.

Comment (palemieux) [reply spec/imsc-hrm.html:437]: The presentation compositor is defined in the context of the operations it performs and I do not think it is necessary (or useful) to create a separate definition for it -- which would also mean creating a definition for all the processing blocks of the HRM.

Comment (nigelmegitt) [reply spec/imsc-hrm.html:437]: That's a good point - maybe a standalone sentence along the lines:

>Each of the terms Presentation Compositor, Glyph Renderer, Glyph Copier, Image Decoder, and Image Copier is defined solely by the algorithmic requirements defined for it in this specification.

would clarify?

Comment (palemieux) [reply spec/imsc-hrm.html:437]: Ok with me. As a note or in the figure caption or ...?

Comment (palemieux) [reply spec/imsc-hrm.html:437]: See https://github.com/w3c/imsc-hrm/pull/61/commits/b03fcfc8557b6f766ad792944d35fb2609079cac

**Comment by nigelmegitt** [comment]
@rhiaro Would this pull request resolve the issue - I think we have likely got consensus for it, at least amongst those who have written and reviewed it.

**Comment by nigelmegitt** [comment]
I'd prefer to hold off on a review until @rhiaro has commented, at least for a few days.

**Review by rhiaro — APPROVED** [review]

**Comment by palemieux** [comment]
@nigelmegitt You apparently need to approve it as well.

**Review by nigelmegitt — APPROVED** [review]

**Pull request merged**